airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/Destination.java

```diff
@@ -29,8 +35,126 @@ AirbyteMessageConsumer getConsumer(JsonNode config,
                                      Consumer<AirbyteMessage> outputRecordCollector)
       throws Exception;
 
+  /**
+   * Default implementation allows us to not have to touch existing destinations while avoiding a lot
+   * of conditional statements in {@link IntegrationRunner}.
+   *
+   * @param config config
+   * @param catalog catalog
+   * @param outputRecordCollector outputRecordCollector
+   * @return AirbyteMessageConsumer wrapped in SerializedAirbyteMessageConsumer to maintain legacy
+   *         behavior.
+   * @throws Exception exception
+   */
+  default SerializedAirbyteMessageConsumer getSerializedMessageConsumer(final JsonNode config,
+                                                                        final ConfiguredAirbyteCatalog catalog,
+                                                                        final Consumer<AirbyteMessage> outputRecordCollector)
+      throws Exception {
+    return new ShimToSerializedAirbyteMessageConsumer(getConsumer(config, catalog, outputRecordCollector));
+  }
+
   static void defaultOutputRecordCollector(final AirbyteMessage message) {
     System.out.println(Jsons.serialize(message));
   }
 
+  @Slf4j
+  class ShimToSerializedAirbyteMessageConsumer implements SerializedAirbyteMessageConsumer {
+
+    private final AirbyteMessageConsumer consumer;
+
+    public ShimToSerializedAirbyteMessageConsumer(final AirbyteMessageConsumer consumer) {
+      this.consumer = consumer;
+    }
+
+    @Override
+    public void start() throws Exception {
+      consumer.start();
+    }
+
+    /**
+     * Consumes an {@link AirbyteMessage} for processing.
+     * <p>
+     * If the provided JSON string is invalid AND represents a {@link AirbyteMessage.Type#STATE}
+     * message, processing is halted. Otherwise, the invalid message is logged and execution continues.
```

Comment:
nit: probably useful to add more details why i.e. STATE messages represent application state and we want to fail fast to ensure data correctness.

Reply:
ah - didn't realise this was moved from an existing class.